shanoir-ng-datasets/src/main/java/org/shanoir/ng/vip/controller/ExecutionDataApiController.java

```
@@ -84,4 +131,115 @@ public ResponseEntity<?> getPath(
+        executionMonitoring.setInputDatasets(inputDatasets);
+
+        // Save monitoring in db.
+        final ExecutionMonitoring createdMonitoring = executionMonitoringService.create(executionMonitoring);
+
+        List<ParameterResourcesDTO> parametersDatasets = executionMonitoringService.createProcessingResources(createdMonitoring, execution.getParametersRessources());
+
+        // 3: create Execution on VIP
+        // init headers with the active access token
+
+        execution.setResultsLocation("shanoir:/" + createdMonitoring.getResultsLocation() + "?token=" + authenticationToken + "&refreshToken=" + execution.getRefreshToken() + "&md5=none&type=File");
+
+        String exportFormat = execution.getExportFormat();
+        String extension = ".nii.gz";
+        if ("dcm".equals(execution.getExportFormat())) {
+            extension = ".zip";
+        }
+
+        Map<String, List<String>> parametersDatasetsInputValues = new HashMap<>();
+        for (ParameterResourcesDTO parameterResourcesDTO : parametersDatasets) {
+            parametersDatasetsInputValues.put(parameterResourcesDTO.getParameter(), new ArrayList<>());
+
+            for (String ressourceId : parameterResourcesDTO.getResourceIds()) {
+                String entityName = "resource_id+" + ressourceId + "+" + parameterResourcesDTO.getGroupBy().name().toLowerCase() + extension;
+                String inputValue = "shanoir:/" + entityName + "?format=" + exportFormat + "&datasetId=" + ressourceId
+                 + "&token=" + authenticationToken + "&refreshToken=" + execution.getRefreshToken() + "&md5=none&type=File";
+                parametersDatasetsInputValues.get(parameterResourcesDTO.getParameter()).add(inputValue);
+            }
+        }
+        execution.setInputValues(parametersDatasetsInputValues);
+
+        try {
+            LOG.error(mapper.writeValueAsString(execution));
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
```

I don't think we should throw a runtime exception here.